### PR TITLE
MediaElement

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

MediaElement is a view for playing video and audio. Media that's supported by the underlying platform can be played from the following sources:

- The web, using a URI (HTTP or HTTPS).
- A resource embedded in the platform application, using the ms-appx:/// URI scheme.
- Files that come from the app's local and temporary data folders, using the ms-appdata:/// URI scheme.
- The device's library.

xct-mediaelement

### API Changes ###

Ported the MediaElement API from Xamarin.Forms.

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of main at time of PR
- [x] Changes adhere to coding standard
- [ ] Updated documentation

### CODE REVIEW

**Review by jfversluis — APPROVED** [review]
I think the code was reviewed in length for Forms and for a first version we probably want to make the transition from Forms to this one as easy as possible

**Comment by jfversluis** [comment]
Also added a little transition guide to the wiki: https://github.com/xamarin/XamarinCommunityToolkit/wiki/Features-moved-from-Xamarin.Forms#mediaelement

**Comment by muhaym** [comment]
https://github.com/xamarin/Xamarin.Forms/issues/11412 Is this handled in this PR?

**Comment by jfversluis** [comment]
> [xamarin/Xamarin.Forms#11412](https://github.com/xamarin/Xamarin.Forms/issues/11412) Is this handled in this PR?

No, but it is in #336 :)

**Comment by muhaym** [comment]
> 
> 
> > [xamarin/Xamarin.Forms#11412](https://github.com/xamarin/Xamarin.Forms/issues/11412) Is this handled in this PR?
> 
> No, but it is in #336 :)

I saw that, but kind of confused about MediaElement's presence in XF repo and where the fix will be done, and kind of ETA which we can expect.

**Comment by jfversluis** [comment]
@muhaym MediaElement (and everything that has the experimental status in Forms) will move out of Xamarin.Forms and be brought over to this Toolkit. This fix will happen here.

I'm planning on merging MediaElement and soon after that fix, somewhere this week and release a new pre-release version. I think we should be able to release a stable v1 soon :)

File: XamarinCommunityToolkit/Views/MediaElement/UriTypeConverter.shared.cs
Comment (pictos) [inline XamarinCommunityToolkit/Views/MediaElement/UriTypeConverter.shared.cs:17]: @jsuarezruiz will we use this method or can I remove it?

**Comment by pictos** [comment]
@jfversluis can you mark this PR with the `DO-NOT-MERGE` label? I'll open a PR later targeting the `mediaelement` branch. After that, we can merge this one

**Comment by jfversluis** [comment]
OK, will that PR be here soon @pictos ? Because I don't want to start chaining PRs and keep this waiting all that time. I'd love to merge fast and stabilize then :)

**Comment by pictos** [comment]
@jfversluis will be, I'll open a draft PR that way the team can track my progress.

**Comment by jfversluis** [comment]
DNM until #342 is merged with this one

File: XamarinCommunityToolkit/Views/MediaElement/UriTypeConverter.shared.cs
Comment (jsuarezruiz) [reply XamarinCommunityToolkit/Views/MediaElement/UriTypeConverter.shared.cs:17]: Is not used, we can remove it.

**Comment by jfversluis** [comment]
OK other PRs merged, do not merge is taken off, @jsuarezruiz can you fix the conflicts? I think we should be good then :)

**Comment by jfversluis** [comment]
In fact, I did it for you. Let's get this merged 💪

**Review by jfversluis — APPROVED** [review]

**Pull request merged**
